x/mongo/driver/session/client_session.go

```diff
@@ -230,7 +234,12 @@ func NewClientSession(pool *Pool, clientID uuid.UUID, opts ...*ClientOptions) (*
 // SetServer will check out a session from the client session pool.
 func (c *Client) SetServer() error {
 	var err error
-	c.Server, err = c.pool.GetSession()
+	if c.RequireNew {
+		c.Server, err = newServerSession()
```

Comment:
Can we pass the flag to the pool to alter the behavior there? So users can use the session identically regardless of future changes in the pool.

Reply:
Could you elaborate on this?

Reply:
Same concern as Matt's underneath. Will join the discussion in his thread.